java/flight/src/main/java/org/apache/arrow/flight/ArrowMessage.java

```diff
@@ -386,8 +389,9 @@ public ArrowMessage parse(InputStream stream) {
   }
 
   @Override
-  public void close() throws Exception {
-    AutoCloseables.close(bufs);
+  public void close() {
+    // Don't use AutoCloseables to avoid a checked exception
+    bufs.forEach(ArrowBuf::close);
```

Comment:
Is ArrowBuf close guaranteed not to throw?

Reply:
it is typically a coding bug, not a program condition that would cause it to throw (thus no checked exception). The only reason I know of is an over-release